components/new-contribution-flow/ContributionFlowPublicMessage.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import themeGet from '@styled-system/theme-get';
+import { Field, Form, Formik } from 'formik';
+import { defineMessages, FormattedMessage, injectIntl, useIntl } from 'react-intl';
+import styled from 'styled-components';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+
+import Avatar from '../../components/Avatar';
+import Container from '../../components/Container';
+import { Flex } from '../../components/Grid';
+import StyledButton from '../../components/StyledButton';
+import StyledInputField from '../../components/StyledInputField';
+import StyledTextarea from '../../components/StyledTextarea';
+import { P } from '../../components/Text';
+
+const PUBLIC_MESSAGE_MAX_LENGTH = 20;
+
+// Styled components
+const PublicMessageContainer = styled(Container)`
+  display: flex;
+  flex-direction: column;
+  background: ${themeGet('colors.white.full')};
+  border: 1px solid ${themeGet('colors.black.400')};
+  border-radius: 10px;
+`;
+
+// GraphQL
+const postContributionPublicMessageMutation = gqlV2/* GraphQL */ `
+  mutation NewContributionFlowEditPublicMessage(
+    $FromCollectiveLegacyId: AccountReferenceInput!
+    $ToCollectiveId: AccountReferenceInput!
+    $message: String
+  ) {
+    editPublicMessage(
+      FromCollectiveLegacyId: $FromCollectiveLegacyId
+      ToCollectiveId: $ToCollectiveId
+      message: $message
+    ) {
+      id
+      publicMessage
+    }
+  }
+`;
+
+// Messages
+const messages = defineMessages({
+  publicMessagePlaceholder: {
+    id: 'contribute.publicMessage.placeholder',
+    defaultMessage: 'Motivate others to contribute in 140 characters :) ...',
+  },
+});
+
+const ContributionFlowPublicMessage = ({ collective, stepProfile, publicMessage }) => {
+  const intl = useIntl();
+
+  // GraphQL & data
+  const [postPublicMessage] = useMutation(postContributionPublicMessageMutation, {
+    context: API_V2_CONTEXT,
+  });
+
+  // Formik
+  const initialValues = {
+    publicMessage: publicMessage || '',
+  };
+
+  const submitPublicMessage = async values => {
+    const { publicMessage } = values;
+    const result = await postPublicMessage({
+      variables: {
+        FromCollectiveLegacyId: { legacyId: stepProfile.id },
+        ToCollectiveId: { id: collective.id },
+        message: publicMessage,
+      },
+    });
+    return result;
```

Comment:
Awaiting the promise then returning it is equivalent to:

```es6
return postPublicMessage(...).then(result => result);
```

You can safely return the promise directly:
```suggestion
    return postPublicMessage({
      variables: {
        FromCollectiveLegacyId: { legacyId: stepProfile.id },
        ToCollectiveId: { id: collective.id },
        message: publicMessage,
      },
    });
```